mysql/mysqldump.go

```diff
@@ -410,9 +408,26 @@ func getTypeModsAndID(conv *internal.Conv, columnType string) (string, []int64)
 
 // handleParseError handles error while parsing mysqldump
 // statements and attempts at creating parsable chunk.
-// Error can be due to insert statement or unsupported Spatial
-// datatypes in create statement.
+// Error can be due to insert statement, unsupported Spatial
+// datatypes in create statement or unsupported stored programs.
 func handleParseError(conv *internal.Conv, chunk string, err error, l [][]byte) ([]ast.StmtNode, bool) {
+	// Check error for statements that are not supported by Pingcap parser.
```

Comment:
Suggest rephrasing slightly:
        // Check error for statements that are not supported by Pingcap parser such
        // as delimiter, function, procedures and triggers.
	// If the error is due to a delimiter, we reparse till the chunk
	// contains 2 delimiters, which is the typical way delimiters
        // are used by mysqldump e.g.
	//    DELIMITER ;; - First one redefines delimiter to something unusual
	//    - What follows is the definition of a function, procedure
	//    - or trigger, which can freely use ';' in its body.
	//    DELIMITER ; - Second one restores default delimiter
	// We also handle the case of functions, procedures or triggers
	// without a delimiter statement.

Reply:
Fixed.